tools/flakeguard/golang/golang.go

```diff
@@ -197,3 +226,138 @@ func FilterPackagesWithTests(pkgs []string) []string {
 	}
 	return testPkgs
 }
+
+// SkipTest is a struct that contains the package and name of the test to skip
+type SkipTest struct {
+	Package    string
+	Name       string
+	JiraTicket string
+
+	// Set by SkipTests
+
+	// If the test was already skipped, or there were issues skipping it, this will be false
+	Skipped bool
+
+	// The file and line number of the test that was skipped
+	File string
+	Line int
+}
+
+// SkipTests finds all package/test pairs provided and skips them
+func SkipTests(repoPath string, testsToSkip []*SkipTest) error {
+	packages, err := Packages(repoPath)
+	if err != nil {
+		return err
+	}
+
+	for _, testToSkip := range testsToSkip {
+		var (
+			packageDir        string
+			packageImportPath = testToSkip.Package
+		)
+		for _, pkg := range packages {
+			if pkg.ImportPath == packageImportPath {
+				packageDir = pkg.Dir
+				break
+			}
+		}
+		if packageDir == "" {
+			return fmt.Errorf("directory for package '%s' not found", packageImportPath)
+		}
+
+		log.Debug().
+			Str("package_dir", packageDir).
+			Str("test", testToSkip.Name).
+			Str("package", packageImportPath).
+			Msg("Skipping test")
+
+		err := filepath.Walk(packageDir, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() || !strings.HasSuffix(info.Name(), "_test.go") {
+				return nil
+			}
+
+			fset := token.NewFileSet()
+			fileAst, err := parser.ParseFile(fset, path, nil, parser.ParseComments)
+			if err != nil {
+				return err
+			}
+
+			found := false
+			ast.Inspect(fileAst, func(n ast.Node) bool {
+				fn, ok := n.(*ast.FuncDecl)
+				if !ok || fn.Name.Name != testToSkip.Name {
+					return true
+				}
```

Comment:
Does this work with "sub tests"? For example: https://smartcontract-it.atlassian.net/browse/DX-759

This test name is `TestAddEVMSolanaLaneBidirectional/MCMS_disabled` but the test is actually a sub test and not the function name itself. (https://github.com/smartcontractkit/chainlink/blob/870357a77a614dc50bf9bde93e0676108a290d6e/deployment/ccip/changeset/crossfamily/v1_6/cs_add_evm_solana_lane_test.go#L32-L44)